frontend/src/pages/home/home.tsx

```diff
@@ -136,27 +156,55 @@ export default class HomePage extends React.Component<HomePageProps, HomePageSta
         )
     }
 
+    randomizeArrayWithSeed(unshuffled_arr: any[], seed: string) {
+        let rng = seedrandom(seed);
+        // Schwartzian transform
+        return unshuffled_arr
+            .map((a) => ({sort: rng(), value: a}))
+            .sort((a, b) => a.sort - b.sort)
+            .map((a) => a.value);
+    }
+
+    // We do this because state setting is async and trying to create this in getData yields empty arrays
     compileCardData() {
-        // We do this because state setting is async and trying to create this in getData yields empty arrays
-        let comboCardData: (Message|Artwork|Video)[] = [];
-        // TODO: This can should be more generalized, but generally there will be fewer art than messages
-        const art_cards_length = this.state.artworks.length;
-        const message_cards_length = this.state.messages.length;
-        const video_cards_length = this.state.videos.length;
-        const index_increment_spacing = Math.floor(message_cards_length/art_cards_length);
-
-        for (let msg_index = 0, art_index = 0, video_index = 0; msg_index < message_cards_length; msg_index++) {
-            comboCardData.push(this.state.messages[msg_index]);
-            if (art_index < art_cards_length && msg_index % index_increment_spacing === 0) {
-                comboCardData.push(this.state.artworks[art_index]);
-                art_index++;
-                // Hack this in...
-                if (video_index < video_cards_length) {
-                    comboCardData.push(this.state.videos[video_index]);
-                    video_index++;
-                }
+        let comboCardData: (Message|Artwork|Video|MultiArtwork)[] = [];
+        let main_content_array: any[] = [];
+        let sub_content_array: any[] = [];
+        let multimedia_count: number = this.state.artworks.length + this.state.videos.length + this.state.multiArtworks.length
+        let index_increment_spacing: number;
+        
+        // The higher count of the two types of content gets to determine the sprinkling of the type of content
+        if (multimedia_count > this.state.messages.length) {
+            main_content_array = this.randomizeArrayWithSeed(
+                main_content_array.concat(this.state.multiArtworks, this.state.artworks, this.state.videos),
+                "manotomo",
+            );
+            // TODO: create a randomly seeded version of the main content array
+            sub_content_array = this.state.messages;
+            
+            index_increment_spacing = Math.floor(multimedia_count / this.state.messages.length);
+        } else {
+            main_content_array = this.state.messages;
+            sub_content_array = this.randomizeArrayWithSeed(
+                sub_content_array.concat(this.state.multiArtworks, this.state.artworks, this.state.videos),
+                "manotomo",
+            );
+
+            index_increment_spacing = Math.floor(this.state.messages.length / multimedia_count);
```

Comment:
Can multimedia_count possibly be zero? we should check for that

Reply:
wait, javascript doesn't die when you perform 1/0 and 0/0??? Math.floor(1/0) evaluates to infinity, and somehow nothing crashes, but what the heck? We prob should check anyways just to be safe.

Reply:
that is the power of javascript but year we should check it